### PR TITLE
High contrast theme

### PR DESCRIPTION
**Pull request by aaronshekey**
This PR explores adding high contrast modes in both light and dark modes. We'll likely need to get this in front of some users to see if it's even a net positive.

Interestingly, with a cursory search, it doesn't appear that there's a reliable media query to auto-select high contrast modes 😔

- [x] Syntax highlighting in dark mode
- [x] Notice border colors

### CODE REVIEW

**Comment by dancormier** [comment]
With develop merged in, I think we have a decent path forward here. Long term, I'd love to make the color generation programmatic but that's a project for another time.

I dug around using [Firefox's _contrast loss_ mode](https://developer.mozilla.org/en-US/docs/Tools/Accessibility_inspector/Simulation) to pick out some contrast issues I'd like to resolve before merging:

- [ ] [Banner](https://deploy-preview-556--stacks.netlify.app/product/components/banners/) - Bottom border too light
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Button group](https://deploy-preview-556--stacks.netlify.app/product/components/banners/) - Selected hard to differentiate from unselected
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Label > Status](https://deploy-preview-556--stacks.netlify.app/product/components/labels/#required) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Navigation](https://deploy-preview-556--stacks.netlify.app/product/components/navigation/#default) - Selected state
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Notices > Filled](https://deploy-preview-556--stacks.netlify.app/product/components/notices/#filled) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Notices > Toast](https://deploy-preview-556--stacks.netlify.app/product/components/notices/#toast) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Pagination](https://deploy-preview-556--stacks.netlify.app/product/components/pagination/#example) - Selected state
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Progress bars > Fill alternates](https://deploy-preview-556--stacks.netlify.app/product/components/progress-bars/#fill-alternates) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [ ] [Progress bars > Segmented](https://deploy-preview-556--stacks.netlify.app/product/components/progress-bars/#segmented) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Sidebar widgets - Alternative colors](https://deploy-preview-556--stacks.netlify.app/product/components/sidebar-widgets/#alternative-colors) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] [Tags](https://deploy-preview-556--stacks.netlify.app/product/components/tags/) - Contrast lower in high contrast mode
    <details>
    <summary>Screenshot</summary>

    image

    </details>

- [x] Theme colors - Theme colors aren't modified in high contrast mode

**Comment by aaronshekey** [comment]
This now has #752 in it, and can serve as the PR moving forward.

**TODO**
- [x] The homepage allows for setting a custom theme and high contrast

File: docs/_includes/layouts/home.html [hunk 1]
```diff
@@ -187,6 +187,25 @@
...
+                    </div>
+                </div>
+
+                <div class="s-popover w-auto wmn-initial wmx-initial"
+                    id="theming-popover"
+                    role="menu">
+                    <div class="s-popover--arrow"></div>
+                    <div class="grid fd-column gs12 gsy">
+                        <button class="grid--cell ta-left s-btn__unset fc-black-600 c-pointer js-darkmode-btn">Toggle dark mode</button>
+                        <button class="grid--cell ta-left s-btn__unset fc-black-600 c-pointer js-highcontrast-btn">Toggle high contrast</button>
```
Comment (b-kelly) [inline docs/_includes/layouts/home.html]: Is this supposed to be here? Looks like an accidental partial copy of the popover above (has the same id, etc).

```suggestion
```

File: docs/_includes/layouts/page-nomenu.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
```
Comment (b-kelly) [inline docs/_includes/layouts/page-nomenu.html]: Is this still needed?

**Review by b-kelly — COMMENTED** [review]
Code looks good! Still doing some manual testing + size checking, so not approving juuuust yet.

**Review by b-kelly — COMMENTED** [review]
Here's my size comparison. I ran:

```
> npm run build
> cd dist/css
> gzip -k stacks*.css
```
I then compared the sizes of each resulting `.gz` file as reported by Windows explorer.

**develop branch**
* stacks.css: 52.7 KB (53,981 bytes)
* stacks.min.css: 48.3 KB (49,481 bytes)

**high contrast**
* stacks.css: 56.8 KB (58,206 bytes)
* stacks.min.css: 51.6 KB (52,839 bytes)

This results in a size increase of:
* stacks.css: 4225 bytes, ~8%
* stacks.min.css: 3358 bytes, ~7%

This size increase is kinda heavy, but not exorbitantly so. I'm going to do some quick checks on the outputted bundle to see if I can't pull some tricks to get it a bit smaller.

**Edit**

I made some tweaks to trim back the output as much as I'm able. The bulk of the new bytes comes from the new color variables, and there's no way for me to cut that out without cutting out the entire feature.

New sizes are as follows:

* stacks.css: 55.3 KB (56,651 bytes)
* stacks.min.css: 50.6 KB (51,859 bytes)

This results in a size increase of:
* stacks.css: 2670 bytes, ~5%
* stacks.min.css: 2378 bytes, ~5%

**Review by b-kelly — APPROVED** [review]
This looks awesome! Excellent work. This was a huge undertaking and it looks like you pretty much nailed it (to my non-designer, non-a11y expert eyes).

**Comment by aaronshekey** [comment]
Ran Backstop and didn't see anything unexpected. There were some changes with respect to how colors are documented and examples in popovers, modals, and notices. Calling it! Merging.

**Pull request merged**